Subject: mac80211: allow AP and VLAN modes

This patch allows switching interfaces into AP/VLAN modes using
cfg80211 (nl80211). Don't allow doing it with wext because then
people will just attempt to do it manually (without hostapd) and
complain that it doesn't work.

Signed-off-by: Johannes Berg <johannes@sipsolutions.net>
---
 net/mac80211/cfg.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -35,6 +35,10 @@ nl80211_type_to_mac80211_type(enum nl802
 	case NL80211_IFTYPE_MESH_POINT:
 		return IEEE80211_IF_TYPE_MESH_POINT;
 #endif
+	case NL80211_IFTYPE_AP:
+		return IEEE80211_IF_TYPE_AP;
+	case NL80211_IFTYPE_AP_VLAN:
+		return IEEE80211_IF_TYPE_VLAN;
 	case NL80211_IFTYPE_WDS:
 		return IEEE80211_IF_TYPE_WDS;
 	default:
